tools/k8s/fetch-creds.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/sh
+
+set -o pipefail
+set -e
+
+prog="$0"
+
+print_help () {
+    echo "usage: $0 [-h|--help] [-f|--force] OUTDIR SERVICE_ACCOUNT [CONTEXT]"
+    echo ""
+    echo "This script uses kubectl to fetch suitable credentials for a"
+    echo "particular service account, which are stored in OUTDIR."
+    echo ""
+    echo "If CONTEXT is not provided, the default k8s context will be used."
+    echo ""
+    echo "The resulting OUTDIR will look like:"
+    echo ""
+    echo "    OUTDIR"
+    echo "    ├── ca.crt           # a file expected by InClusterConfig()"
+    echo "    ├── token            # a file expected by InClusterConfig()"
+    echo "    ├── docker-env-file  # for docker run --env-file"
+    echo "    ├── server           # for k8s apps without InClusterConfig()"
+    echo "    └── configstr        # for caching credentials"
+    echo ""
+    echo "Afterwards, there are two recommended ways to use OUTDIR:"
+    echo ""
+    echo "  - Modify a k8s application to read from OUTDIR directly"
+    echo "    (for example, determined-master does this)"
+    echo ""
+    echo "  - Run an unmodified k8s application via docker run:"
+    echo ""
+    echo "      docker run \\"
+    echo "          -v OUTDIR:/var/run/secrets/kubernetes.io/serviceaccount \\"
+    echo "          --env-file OUTDIR/docker-env-file \\"
+    echo "          my_image_name"
+}
+
+force=n
+outdir=""
+account=""
+context=""
+
+# Manually parse args since getopt varies across unicies.
```

Comment:
is unicies a real thing?

Reply:
Yes, like the word "vertex" -> "vertices", though I think I included an extra 'i'.